### PR TITLE
Backup: accept HEAD requests on /container/restore route

### LINKED ISSUE #1090: Add an API to check current state of a restoration operation
At least, API response should report how many data was been consumed

### PR DESCRIPTION
**Pull request by murlock**
Will closes #1090 
 Also add parameter to control Redis timeout (default is still 24 hours)

### CODE REVIEW

File: oio/container/backup.py [hunk 1]
```diff
@@ -629,6 +632,19 @@ def on_dump(self, req):
...
+    def _do_put_head(self, req, account, container):
+        results = self.redis.get("restore:%s:%s" % (account,
+                                                    container))
+        if not results:
+            return UnprocessableEntity("No restoration in progress")
+        results = json.loads(results)
+        blocks = sum([i['blocks'] for i in results['manifest']])
+        return Response(headers={
+            'X-TarSize': blocks * BLOCKSIZE,
+            'X-TarConsumed': results['end'] * BLOCKSIZE,
```
Comment (fvennetier) [inline oio/container/backup.py]: `X-ConsumedSize` would be clearer.

File: oio/container/backup.py [hunk 2]
```diff
@@ -401,6 +402,8 @@ def __init__(self, conf):
             Rule('/v1.0/container/restore', endpoint='restore'),
         ])
         self.logger = get_logger(self.conf, name="ContainerBackup")
+        self.CACHE = self.conf.get("redis_cache_timeout", self.CACHE)
```
Comment (jfsmig) [inline oio/container/backup.py]: He's back! :man_dancing:

File: oio/container/backup.py (hunk 1 above)
Comment (murlock) [inline oio/container/backup.py]: Done

File: oio/container/backup.py [hunk 3]
```diff
@@ -631,6 +632,19 @@ def on_dump(self, req):
         return Response("Not supported", 405)
 
     @redis_cnx
+    def _do_put_head(self, req, account, container):
+        results = self.redis.get("restore:%s:%s" % (account,
+                                                    container))
+        if not results:
+            return UnprocessableEntity("No restoration in progress")
+        results = json.loads(results)
+        blocks = sum([i['blocks'] for i in results['manifest']])
```
Comment (murlock) [inline oio/container/backup.py]: Indeed

File: oio/container/backup.py (hunk 2 above)
Comment (murlock) [inline oio/container/backup.py]: Renamed as REDIS_TIMEOUT

File: oio/container/backup.py (hunk 1 above)
Comment (jfsmig) [inline oio/container/backup.py]: `X-Consumed-Size` would be HTTP compliant :)

**Pull request merged**